File: frontend/src/modules/purchase-receipts/types/purchase-receipt.ts
export type RecepcionCompraEstado = 'pendiente' | 'completada' | 'cancelada'

export interface RecepcionCompraDetalleCreate {
  producto_id: number
  cantidad_recibida: number
  precio_unitario: number
}

export interface DetalleItem extends RecepcionCompraDetalleCreate {
  id: number
  producto_nombre?: string
  cantidad_solicitada: number
  subtotal: number
}

export interface RecepcionCompraDetalle extends RecepcionCompraDetalleCreate {
  id: number
  recepcion_id: number
}

export interface RecepcionCompraCreate {
  orden_compra_id: number
  proveedor_id: number
  usuario_id: number
  observaciones?: string | null
  estado?: RecepcionCompraEstado
  detalles: RecepcionCompraDetalleCreate[]
}

export interface RecepcionCompra {
  id: number
  orden_compra_id: number
  proveedor_id: number
  usuario_id: number
  fecha: string
  observaciones: string | null
  estado: RecepcionCompraEstado
  created_at: string
  updated_at: string
  detalles: RecepcionCompraDetalle[]
}

export interface RecepcionCompraFilter {
  estado?: RecepcionCompraEstado
}

File: backend/app/schemas/purchase_receipt.py
from datetime import datetime
from enum import Enum
from pydantic import BaseModel, ConfigDict


class RecepcionCompraEstado(str, Enum):
    pendiente = 'pendiente'
    completada = 'completada'
    cancelada = 'cancelada'


class RecepcionCompraDetalleCreate(BaseModel):
    producto_id: int
    cantidad_recibida: int
    precio_unitario: int


class RecepcionCompraDetalleResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    recepcion_id: int
    producto_id: int
    cantidad_recibida: int
    precio_unitario: int


class RecepcionCompraCreate(BaseModel):
    orden_compra_id: int
    proveedor_id: int
    usuario_id: int
    observaciones: str | None = None
    estado: RecepcionCompraEstado = RecepcionCompraEstado.pendiente
    detalles: list[RecepcionCompraDetalleCreate]


class RecepcionCompraResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    orden_compra_id: int
    proveedor_id: int
    usuario_id: int
    fecha: datetime
    observaciones: str | None
    estado: RecepcionCompraEstado
    created_at: datetime
    updated_at: datetime
    detalles: list[RecepcionCompraDetalleResponse] = []
